src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -323,19 +350,55 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
     }
   }
 
+  /** conversion utility for protobuf Timestamp difference to java.time.Duration */
+  private static Duration between(Timestamp from, Timestamp to) {
+    return Duration.ofNanos(Durations.toNanos(Timestamps.between(from, to)));
+  }
+
+  private static void spawnMetricsAccounting(SpawnMetrics.Builder spawnMetrics, ExecutedActionMetadata executionMetadata) {
+    if (!executionMetadata.getWorker().isEmpty()) {
+      // Accumulate queueTime from any previous attempts
+      Duration remoteQueueTime = spawnMetrics.build().remoteQueueTime().plus(
```

Comment:
I have a hard time understanding why you are summing up the queue times. Can you explain this a bit more please?

Reply:
Yes, that was based on the commentary in the SpawnMetrics class: the remoteQueueTime getter in SpawnMetrics has a comment explaining this [here](https://github.com/bazelbuild/bazel/blob/bca23e8bb9fc220a6c523a59b2aa88979deb7981/src/main/java/com/google/devtools/build/lib/actions/SpawnMetrics.java#L110-L113)